src/terminator/mod.rs

```diff
@@ -853,12 +855,50 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             "sysconf" => {
                 let name = self.value_to_primval(args[0], usize)?.to_u64()?;
                 trace!("sysconf() called with name {}", name);
-                let result = match name {
-                    30 => PrimVal::Bytes(4096), // _SC_PAGESIZE
-                    70 => PrimVal::from_i128(-1), // _SC_GETPW_R_SIZE_MAX
-                    _ => return Err(EvalError::Unimplemented(format!("Unimplemented sysconf name: {}", name)))
-                };
-                self.write_primval(dest, result, dest_ty)?;
+                // cache the sysconf integers
+                let paths = &[
+                    (&["libc", "_SC_PAGESIZE"], PrimVal::Bytes(4096)),
+                    (&["libc", "_SC_GETPW_R_SIZE_MAX"], PrimVal::from_i128(-1)),
+                ];
+                let mut result = None;
+                for &(path, path_value) in paths {
+                    if let Ok(instance) = self.resolve_path(path) {
+                        use lvalue::{Global, GlobalId};
+                        let cid = GlobalId { instance, promoted: None };
+                        let val = match self.globals.get(&cid).map(|glob| glob.value) {
+                            Some(value) => value,
+                            None => {
+                                let mir = self.load_mir(instance.def)?;
+                                self.globals.insert(cid, Global::uninitialized(mir.return_ty));
+                                let cleanup = StackPopCleanup::MarkStatic(false);
+                                let name = ty::tls::with(|tcx| tcx.item_path_str(def_id));
+                                trace!("pushing stack frame for global: {}", name);
+                                let frame = self.stack.len();
+                                self.push_stack_frame(
+                                    instance,
+                                    mir.span,
+                                    mir,
+                                    Lvalue::Global(cid),
+                                    cleanup,
+                                )?;
+                                while self.stack.len() != frame {
```

Comment:
I'll create a helper method that'll get used both here and in the rustc int eval